cmd/root.go

```diff
@@ -152,7 +152,12 @@ func Execute(analyticsTracker analytics.Tracker, version string) {
 	}
 
 	err = rootCmd.Execute()
+	outcome := analytics.SUCCESS
+	if _, tracking := rootCmd.Annotations["tracking"]; tracking {
```

Comment:
Do we need to set the annotation? Would it would to do this?
```
defer func() {
	analytics.SendCommandCompletedEvent(&outcome, analyticsTracker)
}()
```

Reply:
I realized that since we're generating new resources and operations, and we might not have "CLI Command Run" set up to run on all of them. So I'm thinking it doesn't make sense to send the event "CLI Command Completed" unless the previous one does.

Reply:
I wanted to make sure we avoid sending "CLI Command Completed" when "CLI Command Run" hasn't run. 
The reason "CLI Command Completed" always runs is because its in `cmd/root.go`. That was the only way I could get errors to be tracked.

Reply:
That makes sense since we wouldn't want to send a completed event without a run event. What about setting state on the analytics tracker? Then we could call `SendCommandCompletedEvent` but it would be a no-op if the internal boolean value is false.
```
type Client struct {
	ID         string
	HTTPClient *http.Client
    sentRunEvent bool // new field
	wg         sync.WaitGroup
}

func (c *Client) SendCommandRunEvent() {
    c.sentRunEvent = true
}

func (c *Client) SendCommandCompletedEvent(outcome, accessToken, baseURI string) {
    if c.sentRunEvent {
        // make request
    }
}
```

Reply:
That makes sense. The thought occurred to me too and I definitely like the idea of no-op based on the boolean.

Reply:
Done